src/Dialect/ONNX/ElementsAttr/ElementsAttrBuilder.hpp

```diff
@@ -50,6 +50,20 @@ class ElementsAttrBuilder {
   static mlir::DenseElementsAttr toDenseElementsAttr(
       mlir::ElementsAttr elements);
 
+  // Compares contents for equality. Argument shapes must be broadcast
+  // compatible. Element types must the same.
+  // Asserts if these preconditions are violated (doesn't return false as that
+  // would hide whether the lhs and rhs are different or incompatible).
```

Comment:
I discoverd that it didn't work to allow element types with different bit widths, because then you get different results depending on which element type you use for the comparison if one is bool and another is a wider unsigned integer type

I also concluded that it's bad to compare at higher precision, so comparison now happens at the type specified by the element type and I added a unit test to demonstrate that unequal f32s can become equal when cast to f16

@AlexandreEichenberger PTAL